x-pack/platform/plugins/shared/entity_manager/server/lib/v2/queries/utils.test.ts

```diff
@@ -176,43 +192,90 @@ describe('mergeEntitiesList', () => {
           'entity.last_seen_timestamp': '2024-11-20T18:00:00.000Z',
           'entity.type': 'service',
           'entity.display_name': 'foo',
+          'service.name': 'foo',
           'host.name': 'host-1',
         },
         {
           'entity.id': 'foo',
           'entity.last_seen_timestamp': '2024-11-20T20:00:00.000Z',
           'entity.type': 'service',
           'entity.display_name': 'foo',
+          'service.name': 'foo',
           'host.name': 'host-2',
         },
         {
           'entity.id': 'foo',
           'entity.last_seen_timestamp': '2024-11-20T16:00:00.000Z',
           'entity.type': 'service',
           'entity.display_name': 'foo',
+          'service.name': 'foo',
           'host.name': ['host-1', 'host-2'],
         },
       ];
 
-      const mergedEntities = mergeEntitiesList(
-        [
-          {
-            metadata_fields: ['host.name'],
-          },
-          {
-            metadata_fields: ['host.name'],
-          },
+      const mergedEntities = mergeEntitiesList({
+        entities,
+        metadataFields: ['host.name'],
+        sources: [
+          { identity_fields: ['service.name'], metadata_fields: [] as string[] },
+          { identity_fields: ['service.name'], metadata_fields: [] as string[] },
         ] as EntitySourceDefinition[],
-        entities
-      );
+      });
       expect(mergedEntities.length).toEqual(1);
       expect(mergedEntities[0]).toEqual({
         'entity.id': 'foo',
         'entity.last_seen_timestamp': '2024-11-20T20:00:00.000Z',
         'entity.type': 'service',
         'entity.display_name': 'foo',
+        'service.name': 'foo',
         'host.name': ['host-1', 'host-2'],
       });
     });
+
+    it('assigns all identity fields to the merged entity', () => {
```

Comment:
I think the point is to not add `entity.identity_fields` listing all found fields as this would not be usable, but setting these fields on the entity will help building the filters (see https://github.com/elastic/elastic-entity-model/issues/209#issuecomment-2528692518)